### DISCUSSION TITLE
How to register IRequestPostProcessor in 12.1.X

### DISCUSSION THREAD

**Discussion by callenj** [LuckyPennySoftware/MediatR#928] [Q&A] [answered]
First i apologize if this is a basic thing, but i have tried to find an example and have tested a few different ways to register to no avail.  Have read the docs a few times and nothing i try seems to work.  Prior to 12.1, this worked fine with automatic registration via scanning.  I understand that from 12.1 on, manual registration is required for Pre/Post processors.

I have a LoginQuery with this signature:
`public record LoginQuery(
    string Username,
    string Password) : ILoginQuery, IRequest<ErrorOr<LoginQueryResult>>;`

I want my post processor to trigger only for these specific queries (requests).  My IRequestPostProcessor signature is:
`public class LoginQueryPostProcessor<TRequest, TResponse> : IRequestPostProcessor<TRequest, TResponse>
    where TRequest : ILoginQuery where TResponse : IErrorOr`

I have tried a few ways to register this, but the post processor is never called when using 12.1.X.  Any examples or advise would be very much appreciated.

**Comment by jbogard** [comment, marked as answer]
How did you register the post processor? That looks like an open post processor so it should be:

```csharp
cfg.AddOpenRequestPostProcessor(typeof(LoginQueryPostProcessor<,>));
```

**Reply by callenj** [reply]
Thank you so much Jimmy.  That worked.  I guess i did not read the docs carefully enough.  I did not try the "Open" method.  (AddOpenRequestPostProcessor).  Really appreciate your work and help.  Happy Friday.

**Reply by jbogard** [reply]
Probably could use better messages around that.

**Comment by cheng93** [comment]
@jbogard 

after moving [MediatR.Extensions.Microsoft.DependencyInjection](https://github.com/jbogard/MediatR.Extensions.Microsoft.DependencyInjection) into this repository.

It does feel like this is a breaking change should of probably gone to 13?

**Comment by seruminar** [comment]
@jbogard Agree with @cheng93, without carefully reading the notes and/or having a thorough dependency testing, it is possible to miss that `IRequestPreProcessor` implementations are no longer automatically registered. This was absolutely a breaking change and not a minor release. If this was requested by someone, perhaps it would help ease into the new API by having an opt-out feature for scanned registrations.
